src/grasshopper/lib/fixtures/__init__.py

```diff
@@ -380,25 +387,51 @@ def pytest_collect_file(parent, path):
 class YamlScenarioFile(pytest.File):
     """The logic behind what to do when a Yaml file is specified in pytest."""
 
+    composite_weighted_user_classes = {}
+
     def collect(self):
         """Collect the file, knowing the path via self.fspath."""
         # Third Party
         import yaml
 
-        raw = yaml.safe_load(self.fspath.open())
+        full_scenarios_list = yaml.safe_load(self.fspath.open())
         valid_scenarios = _get_tagged_scenarios(
-            raw_yaml_dict=raw, config=self.config, fspath=self.fspath
+            full_scenarios_list=full_scenarios_list,
+            config=self.config,
+            fspath=self.fspath,
         )  # tag filter
         for scenario_name, scenario_contents in valid_scenarios.items():
             test_file_name = scenario_contents.get("test_file_name")
+            child_scenarios = scenario_contents.get("child_scenarios")
             if test_file_name:
                 yield Scenario.from_parent(
                     self, name=scenario_name, spec=scenario_contents
                 )
+            elif child_scenarios:
+                import pathlib
+
+                parent_path = pathlib.Path(__file__).parent.resolve()
+                YamlScenarioFile.composite_weighted_user_classes = (
+                    _get_composite_weighted_user_classes(
+                        full_scenarios_list, scenario_contents
+                    )
+                )
+                import shutil
+
+                shutil.copy2(
+                    f"{parent_path}/../journeys/composite_journey.py",
+                    f"{os.getcwd()}/composite_journey.py",
```

Comment:
i would only calculate the destination once, this is the kind of thing it's easy to change in one place and not the other. also might not be a bad idea to log.debug of the file copy parameters.
and lastly, i think shutil.copy2 has a way to specify what to do if the destination file already exists? i assume in this case, we would want to delete and copy a fresh copy?

Reply:
changed to be a variable, `temp_gh_file_path`

And yeah the default behavior of `shutil.copy2` is if the `temp_gh_file_path` file already exists, it will be replaced with the contents of the source file from grasshopper's dir.